### PR TITLE
list instances with stats

### PR DESCRIPTION
**Pull request by csquared**
instance listing include ec2 id, ip address, cpu, and memory

```
$ convox instances
ID          IP              STATUS  PROCESSES  CPU    MEM
i-3963798e  54.85.115.31    active  0          0.00%  0.00%
i-894d0837  54.165.253.255  active  0          0.00%  0.00%
i-c6a72b76  52.71.252.224   active  1          0.00%  19.16%
```

### CODE REVIEW

File: api/controllers/instances.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	instances, err := rack.GetInstances()
+
+	if err != nil {
+		return httperr.Server(err)
+	}
+
+	return RenderJson(rw, instances)
+}
+
+func InstanceTerminate(rw http.ResponseWriter, r *http.Request) *httperr.Error {
```
Comment (csquared) [inline api/controllers/instances.go]: will follow this up with `convox terminate` functionality but want to land `convox instances`

**Comment by csquared** [comment]
```
$ cx api get /instances
[
  {
    "agent": true,
    "cpu": 0,
    "id": "i-3963798e",
    "ip": "54.85.115.31",
    "memory": 0,
    "pending": 0,
    "running": 0,
    "status": "active"
  },
  {
    "agent": true,
    "cpu": 0,
    "id": "i-894d0837",
    "ip": "54.165.253.255",
    "memory": 0,
    "pending": 0,
    "running": 0,
    "status": "active"
  },
  {
    "agent": true,
    "cpu": 0,
    "id": "i-c6a72b76",
    "ip": "52.71.252.224",
    "memory": 0.1916,
    "pending": 0,
    "running": 1,
    "status": "active"
  }
]

```

**Comment by csquared** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;E7xMh5oy&#x2F;218-systemwide-cpu-and-memory"> systemwide &quot;cpu&quot; and &quot;memory&quot;</a>

File: api/controllers/routes.go [hunk 2]
```diff
@@ -36,6 +36,8 @@ func NewRouter() (router *mux.Router) {
 	router.HandleFunc("/apps/{app}/ssl", api("ssl.list", SSLList)).Methods("GET")
 	router.HandleFunc("/apps/{app}/ssl", api("ssl.create", SSLCreate)).Methods("POST")
 	router.HandleFunc("/apps/{app}/ssl/{process}/{port}", api("ssl.delete", SSLDelete)).Methods("DELETE")
+	router.HandleFunc("/instances", api("instances.get", InstancesList)).Methods("GET")
+	router.HandleFunc("/instance/{id}", api("instance.delete", InstanceTerminate)).Methods("DELETE")
```
Comment (ddollar) [inline api/controllers/routes.go]: Should be /instances/{id}

File: api/models/instances.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+	Total int `json:"total"`
+	Free  int `json:"free"`
+	Used  int `json:"used"`
+}
+
+func (ir InstanceResource) PercentUsed() float64 {
+	return float64(ir.Used) / float64(ir.Total)
+}
+
+func (s *System) GetInstances() ([]*Instance, error) {
```
Comment (ddollar) [inline api/models/instances.go]: Possible to add some tests around these API endpoints?

**Comment by ddollar** [comment]
API endpoints are not declared in api/manifest.yml

**Comment by ddollar** [comment]
Let's remove the "pending" API attribute and change "running" to "processes" there as well.

**Pull request merged**